presto-main/src/main/java/io/prestosql/operator/exchange/LocalExchangeSinkOperator.java

```diff
@@ -128,7 +129,13 @@ public boolean isFinished()
     @Override
     public ListenableFuture<?> isBlocked()
     {
-        return sink.waitForWriting();
+        if (isBlocked.isDone()) {
```

Comment:
future of sink is already cached

Reply:
Slightly different context, but essentially is the same justification as [my reply](https://github.com/prestosql/presto/pull/6260#discussion_r540921443) for `PartitionedOutputOperator`, except with `LocalExchangeMemoryManager` instead of `OutputBufferMemoryManager`. Admittedly, the problem is not nearly as bad there because it doesn't call into the memory tracking system but synchronizing on each access is still unnecessary.